tests/unit/test_session.py

```diff
@@ -1132,10 +1208,24 @@ def unit_of_work(txn, *args, **kw):
                     metadata=[
                         ("google-cloud-resource-prefix", database.name),
                         ("x-goog-spanner-route-to-leader", "true"),
+                        (
+                            "x-goog-spanner-request-id",
+                            f"1.{REQ_RAND_PROCESS_ID}.{database.NTH_CLIENT.value}.1.3.1",
+                        ),
                     ],
-                )
-            ]
-            * 2,
+                ),
+                mock.call(
+                    request=request,
+                    metadata=[
+                        ("google-cloud-resource-prefix", database.name),
+                        ("x-goog-spanner-route-to-leader", "true"),
+                        (
+                            "x-goog-spanner-request-id",
+                            f"1.{REQ_RAND_PROCESS_ID}.{database.NTH_CLIENT.value}.1.3.2",
```

Comment:
Same here as for the `BeginTransaction` RPC: Transaction retries should not increase the attempt count. Only retries of a single RPC due to `UNAVAILABLE` errors (and other retryable error codes).